src/main/java/uk/gov/pay/api/json/RequestJsonParser.java

```diff
@@ -41,136 +43,123 @@
-        JsonNode fieldNode = node.get(fieldName);
-        String fieldValue = getStringValue(fieldNode, formatError);
-        check(isNotBlank(fieldValue), isFieldMandatory ?
-                aPaymentError(fieldName, CREATE_PAYMENT_MISSING_FIELD_ERROR) :
-                formatError
-        );
-        return fieldValue;
+    private static String validateAndGetLanguage(JsonNode paymentRequest) {
+        return getStringValue(
+                paymentRequest.get(LANGUAGE_FIELD_NAME),
+                aPaymentError(LANGUAGE_FIELD_NAME, CREATE_PAYMENT_VALIDATION_ERROR, LanguageValidator.ERROR_MESSAGE),
+                aPaymentError(LANGUAGE_FIELD_NAME, CREATE_PAYMENT_VALIDATION_ERROR, LanguageValidator.ERROR_MESSAGE));
     }
 
-    private static Integer parseInteger(JsonNode node, String fieldName, boolean isFieldMandatory, PaymentError.Code validationErrorCode, PaymentError.Code missingErrorCode) {
-        JsonNode fieldNode = node.get(fieldName);
-        PaymentError validationPaymentError = aPaymentError(fieldName, validationErrorCode, "Must be a valid numeric format");
-        Integer fieldValue = getIntegerValue(fieldNode, validationPaymentError);
-        check(fieldValue != null, isFieldMandatory ?
-                aPaymentError(fieldName, missingErrorCode) :
-                validationPaymentError
-        );
-        return fieldValue;
+    private static String validateAndGetDescription(JsonNode paymentRequest) {
+        return getStringValue(
+                paymentRequest.get(DESCRIPTION_FIELD_NAME),
+                aPaymentError(DESCRIPTION_FIELD_NAME, CREATE_PAYMENT_VALIDATION_ERROR,"Must be a valid string format"),
+                aPaymentError(DESCRIPTION_FIELD_NAME, CREATE_PAYMENT_MISSING_FIELD_ERROR));
     }
 
-    private static Boolean parseBoolean(JsonNode node, String fieldName, boolean isFieldMandatory, PaymentError.Code validationErrorCode, PaymentError.Code missingErrorCode) {
-        JsonNode fieldNode = node.get(fieldName);
-        PaymentError validationPaymentError = aPaymentError(fieldName, validationErrorCode, "Must be true or false");
-        Boolean fieldValue = getBooleanValue(fieldNode, validationPaymentError);
-        check(fieldValue != null, isFieldMandatory ?
-                aPaymentError(fieldName, missingErrorCode) :
-                validationPaymentError
-        );
-        return fieldValue;
+    private static String validateAndGetReference(JsonNode paymentRequest) {
+        return getStringValue(
+                paymentRequest.get(REFERENCE_FIELD_NAME),
+                aPaymentError(REFERENCE_FIELD_NAME, CREATE_PAYMENT_VALIDATION_ERROR,"Must be a valid string format"),
+                aPaymentError(REFERENCE_FIELD_NAME, CREATE_PAYMENT_MISSING_FIELD_ERROR));
+    }
+    
+    private static String getStringValue(JsonNode jsonNode, PaymentError validationError, PaymentError missingError) {
```

Comment:
perhaps calling this `validateAndGetString` might be more descriptive of what its doing and consistent with other private method names.